ddclient.in

```diff
@@ -2370,6 +2369,42 @@ sub extract_ipv6 {
     return $ip;
 }
 
+######################################################################
+## Regex that matches an IPv6 GUA address. Accepts embedded leading zeros.
+## A Global Unicast Address (GUA) is any address in a range not otherwise
+## reserved.  So rule out any ULA (FC00::/7), multicast (FF00::/8) or
+## link local (FE80::/10) addresses. To simplify match we require that it
+## start with a non-zero /16 prefix. The negative lookahead at the very
+## start is not required but significantly reduces the number of steps
+## required to match (ie, approx 2x faster).
+######################################################################
+my $regex_ipv6_gua = qr/
+    (?<![:.0-9A-Z])         # Not preceded by a character that could be in the address
+    (?!F[CDF][A-F0-9]{2}:)  # Not a multicast or ULA
+    (?!FE[89AB][A-F0-9]:)   # And not a link local
+    (?!0{0,4}:)             # And does not start with zeros
+    (?=[A-F0-9]{1,4}:)      # But does start with something
+    $regex_ipv6             # And is a valid IPv6 address
+/xi;
+
+######################################################################
+## is_ipv6_gua() validates if string is valid IPv6 Global Unicast Address (GUA)
+## with no preceding or trailing spaces/characters, not even line breaks.
```

Comment:
I don't like using the term GUA here because it has a specific meaning that no longer matches the behavior of these functions. How about:

```suggestion
## is_ipv6_global() returns true if the string is a valid IPv6 address that is
## likely to be globally routable, with no preceding or trailing characters.
```